File: src/00/sigma_core/src/ontology_gen/06/pack_structure_intent.rs
#![allow(unused_imports)]
use super::super::L05::*;

pub fn pack_structure_intent(target_type: u32, target_value: u32, locked: bool) -> i32 {
    let mut intent: u32 = target_type | (target_value << 24);
    if locked {
        intent |= 0x80000000;
    }
    intent as i32
}

File: src/_/06/pack_structure_intent.ts
// SSoT: file:///Users/s0fractal/OMEGA/src/ontology/math/pack_structure_intent.md

export function pack_structure_intent(target_type: number, target_value: number, locked: boolean): number {
    let intent = target_type | (target_value << 24);
    if (locked) {
        intent |= 0x80000000;
    }
    return intent | 0;
}
